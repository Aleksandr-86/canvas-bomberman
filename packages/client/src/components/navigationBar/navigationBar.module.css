.navigationBar {
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  gap: 30px;
  align-content: flex-start;
  justify-content: flex-end;
  align-items: center;
  width: inherit;
  padding: 10px 25px 0 0;
  font-size: 40px;
  text-decoration: none;
}

.link {
  text-decoration: none;
  color: var(--text-link-color);
}
