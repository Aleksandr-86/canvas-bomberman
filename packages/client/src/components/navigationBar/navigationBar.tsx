import { Link } from 'react-router-dom'

import { useAppDispatch, useAppSelector } from '../../store/hooks'
import { getUser } from '../../store/selectors'
import avatarImg from '../../assets/images/avatar.png'

import styles from './navigationBar.module.css'
import { logout } from '../../store/actions/userActions'

export const NavigationBar = () => {
  const dispatch = useAppDispatch()
  const { login, displayName } = useAppSelector(getUser).user

  return (
    <nav className={styles.navigationBar}>
      <img src={avatarImg} alt="user-avatar" draggable="false" />
      <Link className={styles.link} to="/profile">
        {displayName || login}
      </Link>
      <Link className={styles.link} to="/" onClick={() => dispatch(logout())}>
        Выйти
      </Link>
    </nav>
  )
}
