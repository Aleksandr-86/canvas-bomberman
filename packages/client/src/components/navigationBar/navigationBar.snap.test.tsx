import * as renderer from 'react-test-renderer'
import { NavigationBar } from './navigationBar'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

const mockStore = configureStore([])
const store = mockStore({ user: { displayName: `Alex` } })

describe(`Snapshot test component NavigationBar`, () => {
  test(`Should correct render (with store)`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter initialEntries={[`/`]}>
            <NavigationBar />
          </MemoryRouter>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
