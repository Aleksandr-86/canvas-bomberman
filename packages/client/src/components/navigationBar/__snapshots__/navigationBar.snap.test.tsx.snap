// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test component NavigationBar Should correct render (with store) 1`] = `
<nav
  className="navigationBar"
>
  <img
    alt="user-avatar"
    draggable="false"
  />
  <a
    className="link"
    href="/profile"
    onClick={[Function]}
  >
    Alex
  </a>
  <a
    className="link"
    href="/"
    onClick={[Function]}
  >
    Выйти
  </a>
</nav>
`;
